src/main/java/co/cask/hydrator/plugin/batch/aggreagtor/aggregator/Sampling.java

```diff
@@ -0,0 +1,250 @@
+                while (count < finalSampleSize) {
+                    StructuredRecord record = recordArray.get(0);
+                    sampleData.add(getSampledRecord(record, randomValue.nextFloat(), schemaWithRandomField));
+                    count++;
+                }
+
+                while (count < recordArray.size()) {
+                    StructuredRecord structuredRecord = (StructuredRecord) sampleData.get();
+                    Float randomFloat = randomValue.nextFloat();
+                    if ((float) structuredRecord.get("random") < randomFloat) {
+                        sampleData.remove();
+                        StructuredRecord record = recordArray.get(count);
+                        sampleData.add(getSampledRecord(record, randomFloat, structuredRecord.getSchema()));
+                    }
+                    count++;
+                }
+
+                Iterator<StructuredRecord> sampleDataIterator = sampleData.iterator();
+                while (sampleDataIterator.hasNext()) {
+                    StructuredRecord sampledRecord = sampleDataIterator.next();
+                    StructuredRecord.Builder builder = StructuredRecord.builder(inputSchema);
+                    for (Schema.Field field : sampledRecord.getSchema().getFields()) {
+                        if (!field.getName().equalsIgnoreCase("random")) {
+                            builder.set(field.getName(), sampledRecord.get(field.getName()));
+                        }
+                    }
+                    emitter.emit(builder.build());
+                }
+                break;
+        }
+    }
+
+    public StructuredRecord getSampledRecord(StructuredRecord record, Float random, Schema schema) {
+        StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+        for (Schema.Field field : record.getSchema().getFields()) {
+            builder.set(field.getName(), record.get(field.getName()));
+        }
+        builder.set("random", random);
+        return builder.build();
+    }
+
+    /**
+     * Builds the schema for Reservoir sampling algorithm. Adding field for random value.
+     */
+    private Schema createSchemaWithRandomField(Schema inputSchema) {
+        List<Schema.Field> fields = new ArrayList<>();
+        fields.add(Schema.Field.of("random", Schema.of(Schema.Type.FLOAT)));
+        for (Schema.Field field : inputSchema.getFields()) {
+            fields.add(field);
+        }
+        return Schema.recordOf("schema", fields);
+    }
+
+    /**
+     * Config for Sampling Plugin.
+     */
+    public static class SamplingConfig extends PluginConfig {
+
+        @Nullable
+        @Description("The number of records that needs to be sampled from the input records.")
+        @Macro
+        private Integer sampleSize;
+
+        @Nullable
+        @Description("The percenatage of records that needs to be sampled from the input records.")
```

Comment:
please add: Either of 'samplePercentage' or 'sampleSize' should be specified for this plugin.